plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -294,13 +301,16 @@ private static ReaderPageSource createOrcPageSource(
 
             OrcReader reader = OrcReader.createOrcReader(orcDataSource, options)
                     .orElseThrow(() -> new TrinoException(ICEBERG_BAD_DATA, "ORC file is zero length"));
+
             List<OrcColumn> fileColumns = reader.getRootColumn().getNestedColumns();
-            Map<Integer, OrcColumn> fileColumnsByIcebergId = mapIdsToOrcFileColumns(fileColumns);
-            Map<String, OrcColumn> fileColumnsByName = null;
-            if (fileColumnsByIcebergId.isEmpty()) {
-                fileColumnsByName = uniqueIndex(fileColumns, orcColumn -> orcColumn.getColumnName().toLowerCase(ENGLISH));
+            if (nameMapping.isPresent() && !hasIds(reader.getRootColumn())) {
+                fileColumns = fileColumns.stream()
+                        .map(orcColumn -> setMissingFieldIds(orcColumn, nameMapping.get(), ImmutableList.of(orcColumn.getColumnName())))
```

Comment:
It is case sensitive. Looks like ORC has the correct case here, but the Parquet metadata reader looses the column casing so that's not working for mixed-case columns.

Reply:
Had to add some methods to convert the NameMapping to lowercase for Parquet to work properly. Modified one of the test cases to have mixed case column names.

Reply:
thanks for investigating and fixing this. Even for ORC, looks like we still need to force casing in field mapper since the (shared) struct reader also does lowercasing.